sdk/loadtestservice/load-testing-rest/src/beginCreateOrUpdateTestRun.ts

```diff
@@ -0,0 +1,150 @@
+
+/**
+ * Creates a poller to poll for test run status.
+ * @param client - The Load Testing client.
+ * @param options - The operation options.
+ * @returns A poller which can be called to poll until completion of the job.
+ */
+export async function beginCreateOrUpdateTestRun(
+  client: AzureLoadTestingClient,
+  testRunId: string,
+  testRunParams: TestRunCreateOrUpdateParameters,
+  polledOperationOptions: PolledOperationOptions = {}
+): Promise<TestRunStatusPoller> {
+  // Creating the test run
+  const testRunCreationResult = await client
+    .path("/test-runs/{testRunId}", testRunId)
+    .patch(testRunParams);
+
+  if (isUnexpected(testRunCreationResult)) {
+    throw testRunCreationResult.body.error;
+  }
+
+  if (testRunCreationResult.body.testRunId === undefined)
+    throw new Error("Test Run ID returned as undefined.");
+
+  type Handler = (state: OperationState<TestRunGet200Response>) => void;
+
+  const state: OperationState<TestRunGet200Response> = {
+    status: "notStarted",
+  };
+
+  const progressCallbacks = new Map<symbol, Handler>();
+  const processProgressCallbacks = async (): Promise<void> =>
+    progressCallbacks.forEach((h) => h(state));
+  let resultPromise: Promise<TestRunGet200Response> | undefined;
+  let cancelJob: (() => void) | undefined;
+  const abortController = new AbortController();
+  const currentPollIntervalInMs = polledOperationOptions.updateIntervalInMs ?? 2000;
+
+  const poller: SimplePollerLike<OperationState<TestRunGet200Response>, TestRunGet200Response> = {
+    async poll(_options?: { abortSignal?: AbortSignalLike }): Promise<void> {
+      let getTestRunResult = await client.path("/test-runs/{testRunId}", testRunId).get();
+      if (isUnexpected(getTestRunResult)) {
+        throw getTestRunResult.body.error;
+      }
+
+      if (getTestRunResult.body.status === "FAILED") {
+        state.status = "failed";
+        state.error = new Error(getTestRunResult.body.status);
+      }
+
+      if (
+        getTestRunResult.body.status === "CANCELLING" ||
+        getTestRunResult.body.status === "CANCELLED"
+      ) {
+        state.status === "canceled";
+      }
+
+      if (getTestRunResult.body.status === "DONE") {
+        state.status = "succeeded";
+        state.result = getTestRunResult;
+      }
+
+      await processProgressCallbacks();
+
+      if (state.status === "canceled") {
+        throw new Error("Operation was canceled");
+      }
+      if (state.status === "failed") {
+        throw state.error;
+      }
+    },
+
+    pollUntilDone(pollOptions?: { abortSignal?: AbortSignalLike }): Promise<TestRunGet200Response> {
+      return (resultPromise ??= (async () => {
+        const { abortSignal: inputAbortSignal } = pollOptions || {};
+        const { signal: abortSignal } = inputAbortSignal
+          ? new AbortController([inputAbortSignal, abortController.signal])
+          : abortController;
+        if (!poller.isDone()) {
+          await poller.poll({ abortSignal });
```

Comment:
This call takes `abortSignal` as input but `poll` ignores it. I think we will need to handle the signal manually in `poll`, e.g.
https://github.com/Azure/azure-sdk-for-js/blob/9d3364c8c5b5ec9cb62989dbf0ee872a63708098/sdk/core/core-rest-pipeline/src/nodeHttpClient.ts#L78-L81

Reply:
Updated the poll function.